src/builders/oracle.ts

```diff
@@ -34,18 +35,29 @@ export const updateOracles = async (
     const priceIds = assetCoinNames.map((assetCoinName) =>
       builder.address.get(`core.coins.${assetCoinName}.oracle.pyth.feed`)
     );
-    const pythConnection = new SuiPriceServiceConnection(
-      builder.isTestnet
-        ? 'https://hermes-beta.pyth.network'
-        : 'https://hermes.pyth.network'
-    );
-    const priceUpdateData =
-      await pythConnection.getPriceFeedsUpdateData(priceIds);
-    await pythClient.updatePriceFeeds(
-      txBlock.txBlock,
-      priceUpdateData,
-      priceIds
-    );
+
+    // iterate through the priceIds and update the price feeds
+    const endpoints = PYTH_ENDPOINTS[builder.isTestnet ? 'testnet' : 'mainnet'];
+    for (const endpoint of endpoints) {
+      try {
+        const pythConnection = new SuiPriceServiceConnection(endpoint);
+        const priceUpdateData =
+          await pythConnection.getPriceFeedsUpdateData(priceIds);
+        await pythClient.updatePriceFeeds(
+          txBlock.txBlock,
+          priceUpdateData,
+          priceIds
+        );
+
+        return;
```

Comment:
- Iterate through `endpoints` list and immediately return on success `priceUpdateData` and `updatePriceFeeds`, else try next endpoint